cmd/replay/main.go

```diff
@@ -46,6 +47,19 @@ func getBlockHeaderByHash(hash common.Hash) *types.Header {
 	return header
 }
 
+func getHotShotHeaderByHash(hash common.Hash) *espresso.Header {
+	enc, err := wavmio.ResolveTypedPreimage(arbutil.Keccak256PreimageType, hash)
+	if err != nil {
+		return nil
+	}
+	var header &espresso.Header
+	if err := json.Unmarshal(header, &enc); err != nil {
```

Comment:
I'm confused by this. If the preimage we fetched is a JSON object, that means `hash` is the keccak hash of a JSON string. But usually we don't have the keccak hash of the header serialized as JSON. Usually we have the Hotshot block commitment, which is an incremental keccak hash of all the fields in the header

Reply:
Yes still trying to wrap my head around this preimage business, as @sveitser  suggested, might be better entirely to treat hotshot commitments like delayed inbox messages